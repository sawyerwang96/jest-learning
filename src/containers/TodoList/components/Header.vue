<template>
  <div class="headerWrapper">
    <div class="headerInner">
      <span class="des">TodoList</span>
      <input
        class="todoInput"
        type="text"
        data-test="header-input"
        v-model="inputValue"
        @keyup.enter="add"
      />
    </div>
    <!-- <el-row type="flex" class="row-bg" justify="center">
      <el-col :span="4">
        <div class="grid-content bg-purple title">
          <h2>TodoList</h2>
        </div>
      </el-col>
      <el-col :span="8">
        <div class="grid-content bg-purple">
          TodoList
          <input
            type="text"
            data-test="input"
            v-model="inputValue"
            @keyup.enter="add"
          />
          <el-input
            id="todoInput"
            data-test="input"
            v-model="inputValue"
            @change="add"
            placeholder="添加ToDo"
          ></el-input>
        </div>
      </el-col>
    </el-row> -->
  </div>
</template>

<script>
export default {
  data () {
    return {
      inputValue: ''
    }
  },
  methods: {
    add () {
      if (this.inputValue) {
        this.$emit('add', this.inputValue)
        this.inputValue = ''
      }
    }
  }
}
</script>

<style scoped lang="scss">
.headerWrapper {
  background-color: #666;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 60px;

  .headerInner {
    display: flex;
    justify-content: space-between;
    width: 600px;
  }

  .des {
    font-size: 30px;
    color: #FFFFFF;
  }

  .todoInput {
    width: 400px;
    padding-left: 10px;
    font-size: 18px;
  }
}
</style>
