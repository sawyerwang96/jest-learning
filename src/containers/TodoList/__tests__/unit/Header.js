import { shallowMount } from '@vue/test-utils'
import Header from '@/containers/TodoList/components/Header.vue'
import { findTestWrapper } from '@/utils/test.util.js'
it('Header style change', () => {
  const wrapper = shallowMount(Header)
  expect(wrapper).toMatchSnapshot()
})

it('Header 包含input框', () => {
  const wrapper = shallowMount(Header)
  // const input = wrapper.find('[data-test="input"]')
  const input = findTestWrapper(wrapper, 'input')
  expect(input.exists()).toBe(true)
})

it('Header input初始内容为空', () => {
  const wrapper = shallowMount(Header)
  const inputValue = wrapper.vm.$data.inputValue
  expect(inputValue).toBe('')
})

it('Header input框的value发生变化,数据应该跟着编', () => {
  const wrapper = shallowMount(Header)
  // const input = wrapper.find('[data-test="input"]')
  const input = findTestWrapper(wrapper, 'input')
  input.setValue('sawyer')
  const inputValue = wrapper.vm.$data.inputValue
  expect(inputValue).toBe('sawyer')
})

it('Header input框按下enter键,inputValue为空时,无反应', () => {
  const wrapper = shallowMount(Header)
  // const input = wrapper.find('[data-test="input"]')
  const input = findTestWrapper(wrapper, 'input')
  input.setValue('')
  input.trigger('keyup.enter')
  expect(wrapper.emitted().add).toBeFalsy()
})

it('Header input框按下enter键,inputValue为不空时,向外触发add事件,同时清空inputValue', () => {
  const wrapper = shallowMount(Header)
  // const input = wrapper.find('[data-test="input"]')
  const input = findTestWrapper(wrapper, 'input')
  input.setValue('haha')
  input.trigger('keyup.enter')
  expect(wrapper.emitted().add).toBeTruthy()
  const inputValue = wrapper.vm.inputValue
  expect(inputValue).toBe('')
})
