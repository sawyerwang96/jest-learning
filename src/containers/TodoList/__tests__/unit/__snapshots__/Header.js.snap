// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue style change 1`] = `
<div class="headerWrapper">
  <div class="headerInner"><span class="des">TodoList</span> <input type="text" data-test="input" class="todoInput"></div>
</div>
`;
